fili-security/src/main/java/com/yahoo/bard/webservice/web/security/RoleBasedRoutingRequestMapper.java

```
@@ -0,0 +1,69 @@
+    RequestMapper<T> defaultMapper;
+
+    /**
+     * Constructor.
+     *
+     * @param resourceDictionaries  The dictionaries to use for request mapping.
+     * @param prioritizedRoleBasedMappers  A map of roles to mappers for each role, with a deterministic entry order.
+     * @param defaultMapper The default mapper to apply if no other roles match. (null means finish mapping)
+     */
+    public RoleBasedRoutingRequestMapper(
+            ResourceDictionaries resourceDictionaries,
+            LinkedHashMap<String, RequestMapper<T>> prioritizedRoleBasedMappers,
+            RequestMapper<T> defaultMapper
+    ) {
+        super(resourceDictionaries);
+        this.prioritizedRoleBasedMappers = prioritizedRoleBasedMappers;
+        this.defaultMapper = defaultMapper;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param resourceDictionaries  The dictionaries to use for request mapping.
+     * @param prioritizedRoleBasedMappers  A map of roles to mappers for each role, with a deterministic entry order.
+     */
+    public RoleBasedRoutingRequestMapper(
+            ResourceDictionaries resourceDictionaries,
+            LinkedHashMap<String, RequestMapper<T>> prioritizedRoleBasedMappers
+    ) {
+        this(resourceDictionaries, prioritizedRoleBasedMappers, null);
+    }
+
+    @Override
+    public T apply(T request, ContainerRequestContext context)
+            throws RequestValidationException {
+        SecurityContext securityContext = context.getSecurityContext();
+        RequestMapper<T> mapper = prioritizedRoleBasedMappers.keySet().stream()
+                .filter(securityContext::isUserInRole)
+                .peek(it -> System.out.println(it))
+                .map(prioritizedRoleBasedMappers::get)
+                .peek(it -> System.out.println(it))
+                .findFirst().orElse(defaultMapper);
+        return (mapper == null) ?
+                request :
+                mapper.apply(request, context);
```

I don't think we want to treat a `null` mapper as the identity. This effectively means that if a user doesn't have any of the specified roles, they get FULL ACCESS TO EVERYTHING FOREVER.  People concerned about security usually get twitchy when the default is FULL ACCESS TO EVERYTHING FOREVER.

Instead, we should probably throw an exception here (an `IllegalArgumentException` or something).